gpiozero/output_devices.py

```diff
@@ -503,6 +503,243 @@ def _blink_device(
             self._write(value)
             if self._blink_thread.stopping.wait(delay):
                 break
+            
+class PWMBuzzer(PWMOutputDevice):
+    """
+    Extends :class: `PWMOutputDevice` and represents a piezo or tonal buzzer 
+    class.
+
+    :param: int pin:
+        The GPIO pin which the buzzer is attached to. See :ref:`pin-numbering` for
+        valid pin numbers.
+    
+    :param bool active_high:
+        If ``True`` (the default), the :meth:`on` method will set the GPIO to 
+        HIGH. If ``False``, the :meth:`on` method will set the GPIO to LOW (the
+        :meth:`off` method always does the opposite).
+    
+    :param float initial_value:
+        If ``0`` (the default), the buzzer will be off initially. Other values
+        between 0 and 1 can be specified as an initial value for the buzzer.
+        Note that ``None`` cannot be specified (unlike the parent class) as
+        there is no way to tell PWM not to alter the state of the pin.
+
+    :param float frequency:
+        The frequency (in Hz) of pulses emitted to drive the buzzer. Defaults
+        to 100Hz.
+    
+    :param Factory pin_factory:
+        See :doc:`api_pins` for more information (this is an advanced feature
+        which most users can ignore).
+    """
+
+    def __init__(
+            self, pin=None, active_high=True, initial_value=0, frequency=100,
+            pin_factory=None):
+        super(PWMBuzzer, self).__init__(
+            pin=pin, active_high=active_high, initial_value=initial_value, frequency=frequency,
+            pin_factory=pin_factory
+        )
+
+        self.notes = {
```

Comment:
Part of me wants these dicts to be declared outside of the file. What do you think @waveform80 @lurch?

Reply:
Something like a `notes_data.py`, which `output_devices.py` then imports?

Reply:
That's what I was thinking

Reply:
OK, I've moved the two dicts into `notes_data.py` and added the import to `output_devices.py`